docs/sql-distributed-sql-engine-spark-sql-cli.md

```diff
@@ -0,0 +1,185 @@
+Configuration of Hive is done by placing your `hive-site.xml`, `core-site.xml` and `hdfs-site.xml` files in `conf/`.
+
+## Spark SQL Command Line Options
+
+You may run `./bin/spark-sql --help` for a complete list of all available options.
+
+    CLI options:
+     -d,--define <key=value>          Variable substitution to apply to Hive
+                                      commands. e.g. -d A=B or --define A=B
+        --database <databasename>     Specify the database to use
+     -e <quoted-query-string>         SQL from command line
+     -f <filename>                    SQL from files
+     -H,--help                        Print help information
+        --hiveconf <property=value>   Use value for given property
+        --hivevar <key=value>         Variable substitution to apply to Hive
+                                      commands. e.g. --hivevar A=B
+     -i <filename>                    Initialization SQL file
+     -S,--silent                      Silent mode in interactive shell
+     -v,--verbose                     Verbose mode (echo executed SQL to the
+                                      console)
+
+## The hiverc File
+
+The Spark SQL CLI when invoked without the `-i` option will attempt to load `$HIVE_HOME/bin/.hiverc` and `$HOME/.hiverc` as initialization files.
+
+## Supported comment types
+
+<table class="table">
+<tr><th>Comment</th><th>Example</th></tr>
+<tr>
+  <td>simple comment</td>
+  <td>
+  <code>
+      -- This is a simple comment.
+      <br>
+      SELECT 1;
+  </code>
+  </td>
+</tr>
+<tr>
+  <td>bracketed comment</td>
+  <td>
+    <code>
+        /* This is a bracketed comment. */
+        <br>
+        SELECT 1;
+    </code>
+  </td>
+</tr>
+<tr>
+  <td>nested bracketed comment</td>
+  <td>
+    <code>
+        /*  This is a /* nested bracketed comment*/ .*/
+        <br>
+        SELECT 1;
+    </code>
+  </td>
+</tr>
+</table>
+
+## Spark SQL CLI Interactive Shell Commands
+
+When `./bin/spark-sql` is run without either the `-e` or `-f` option, it enters interactive shell mode.
+Use `;` (semicolon) to terminate commands. Notice:
+1. CLI use `;` to terminate commands only when it's at the end of line and it's not escaped by `\\;`.
+2. `;` is the only way to terminate commands, if user type `SELECT 1` and press enter, console will just wait for input.
+3. If user type multiple commands in one line like `SELECT 1; SELECT 2;`, commands `SELECT 1` and `SELECT 2` will be executed separatly.
+4. If `;` in a simple comment 
+   ```sql
+   -- This is a comment;
+   SELECT 1;
+   ```
+   This comment line will just be ignored. If `;` in a bracketed command and not at the end of line,
+    ```sql
+    /* This is a comment contains ';'. */
+    SELECT 1;
+    ```
+  it will not terminate commands. If `;` in a bracketed command and in the end of line, 
+     ```sql
+     /* This is a comment contains ;
+     */ SELECT 1;
+     ```
+  it will terminate commands into  `/* This is a comment contains ` and `*/ SELECT 1`.
```

Comment:
and we hit failures when executing `/* This is a comment contains `

Reply:
```
spark-sql> /* This is a comment contains ;
Error in query:
Unclosed bracketed comment(line 1, pos 0)

== SQL ==
/* This is a comment contains ;
^^^

spark-sql>      */ SELECT 1;
Error in query:
extraneous input '*/' expecting {'(', 'ADD', 'ALTER', 'ANALYZE', 'CACHE', 'CLEAR', 'COMMENT', 'COMMIT', 'CREATE', 'DELETE', 'DESC', 'DESCRIBE', 'DFS', 'DROP', 'EXPLAIN', 'EXPORT', 'FROM', 'GRANT', 'IMPORT', 'INSERT', 'LIST', 'LOAD', 'LOCK', 'MAP', 'MERGE', 'MSCK', 'REDUCE', 'REFRESH', 'REPLACE', 'RESET', 'REVOKE', 'ROLLBACK', 'SELECT', 'SET', 'SHOW', 'START', 'TABLE', 'TRUNCATE', 'UNCACHE', 'UNLOCK', 'UPDATE', 'USE', 'VALUES', 'WITH'}(line 1, pos 5)

== SQL ==
     */ SELECT 1
-----^^^

spark-sql>
```